### ISSUE TITLE
False positive `expression must have a constant value` with `__builtin_choose_expr` in `_Static_assert`

### ISSUE THREAD

**Issue by sammko** [microsoft/vscode-cpptools#10168]
### Environment

- OS and Version: Arch
- VS Code Version: 1.74.0-insider 96db0b339360bc72d27d869d81a557540ad5f800
- C/C++ Extension Version: v1.13.4 Pre-Release

### Bug Summary and Steps to Reproduce

```cpp
#define ELEMENTSOF(x)                                                    \
        (__builtin_choose_expr(                                          \
                !__builtin_types_compatible_p(typeof(x), typeof(&*(x))), \
                sizeof(x)/sizeof((x)[0]),                                \
                (void)0))

int main(void) {
    int array1[1] = {0};
    _Static_assert(ELEMENTSOF(array1) <= 20, "hi");
    return 0;
}
```

An error "expression must have a constant value" is reported. Changing compiler, c standard and intelli sense mode didn't help. The program compiles under both gcc and clang with `-Wall -Wextra -pedantic`. gcc is quiet, clang gives only a warning

```
xyz.c:9:20: warning: extension used [-Wlanguage-extension-token]
    _Static_assert(ELEMENTSOF(array1) <= 20, "hi");
                   ^
xyz.c:3:47: note: expanded from macro 'ELEMENTSOF'
                !__builtin_types_compatible_p(typeof(x), typeof(&*(x))), \
                                              ^
xyz.c:9:20: warning: extension used [-Wlanguage-extension-token]
xyz.c:3:58: note: expanded from macro 'ELEMENTSOF'
                !__builtin_types_compatible_p(typeof(x), typeof(&*(x))), \
```

Related to #8294

**Comment by sean-mcmanus** [comment]
This is a duplicate of https://github.com/microsoft/vscode-cpptools/issues/9368

**Comment by sammko** [comment]
Is it? Happens with
```
"compilerPath": "/usr/bin/gcc",
"intelliSenseMode": "linux-gcc-x64",
```

Also no function pointers involved?

**Comment by sean-mcmanus** [comment]
The fix should be available when we release 1.14.0 (pre-release) -- not sure when.

**Comment by sean-mcmanus** [comment]
The fix is available with 1.14.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0
